compose/service.py

```diff
@@ -266,20 +274,29 @@ def create_container(self,
 
         return Container.create(self.client, **container_options)
 
-    def ensure_image_exists(self, do_build=True):
+    def ensure_image_exists(self, do_build=BuildAction.none):
+        if self.can_be_built() and do_build == BuildAction.force:
+            self.build()
+            return
+
         try:
             self.image()
             return
         except NoSuchImageError:
             pass
 
-        if self.can_be_built():
-            if do_build:
-                self.build()
-            else:
-                raise NeedsBuildError(self)
-        else:
+        if not self.can_be_built():
             self.pull()
+            return
+
+        if do_build == BuildAction.skip:
+            raise NeedsBuildError(self)
+
+        self.build()
+        log.warn(
+            "Image for service {} was build because it was not found. To "
+            "rebuild this image you must use the `build` command or the "
+            "--build flag.".format(self.name))
```

Comment:
Typo: "was build" -> "was built"

Perhaps "because it did not already exist"?

Also, might be nice to specify the exact commands:

> To rebuild this image, use `docker-compose build` or `docker-compose up --build`.